client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/TopologyInfo.java

```diff
@@ -39,11 +39,11 @@
    private Collection<SocketAddress> servers = new ArrayList<>();
    private Map<byte[], ConsistentHash> consistentHashes = CollectionFactory.makeMap(ByteArrayEquivalence.INSTANCE, AnyEquivalence.getInstance());
    private Map<byte[], Integer> segmentsByCache = CollectionFactory.makeMap(ByteArrayEquivalence.INSTANCE, AnyEquivalence.getInstance());
-   private volatile AtomicInteger topologyId;
+   private Map<byte[], AtomicInteger> topologyIds = CollectionFactory.makeMap(ByteArrayEquivalence.INSTANCE, AnyEquivalence.getInstance());
```

Comment:
Yes, I think so since there is a happens-before relationship between updating a view by a thread and another thread querying the value to send it to the server indicating its own view id. Also, there's now a check to verify that once a thread has set a view id, no other threads that might have received the same view apply it again, and this comparison is done with topology id.